enterprise_access/utils.py

```diff
@@ -245,3 +245,27 @@ def _days_from_now(days_from_now=0, date_format=None):
     if not date_format:
         return date
     return date.strftime(date_format)
+
+
+def get_course_run_metadata_for_assignment(assignment, content_metadata):
+    """
+    Retrieves metadata for a specific course run associated with an assignment. If the assignment has
+    a preferred course run, returns the metadata for that run. If the preferred run metadata is not 
+    found, returns normalized_metadata.
+
+    Args:
+        assignment (dict): The assignment object.
+        content_metadata (dict): The content metadata object.
+
+    Returns:
+        dict: Course run metadata if available, otherwise normalized_metadata.
+    """
+    if preferred_course_run_key := assignment.preferred_course_run_key:
+        course_runs = content_metadata.get('course_runs', [])
+        for course_run in course_runs:
+            if course_run.get('key') == preferred_course_run_key:
```

Comment:
nit: could opt to use `next` to extract the desired course run, if you wanted. might be a bit more "pythonic", e.g.:

```python
course_run = next((run for run in course_runs if run.get('key') == preferred_course_run_key), None)
if not course_run:
    logger.warning(
        'Course run metadata not found for preferred course run key %s in content metadata %s', 
        preferred_course_run_key,
        content_metadata.get('key'),
    )
    return course_run
```

(Opted to add logging when the `preferred_course_run_key` is not found in the list of `course_runs`).